File: VacationCost/C#/VacationCost/VacationCostCalculator.cs
using System;

namespace VacationCost
{
    public class VacationCostCalculator
    {
        public double DistanceToDestination { get; set; }

        public decimal CostOfVacation(string transportMethod)
        {
            switch (transportMethod)
            {
                case "Car":
                    return (decimal) (DistanceToDestination * 1);
                case "Plane":
                    return (decimal) (DistanceToDestination * 2);
                default:
                    throw new ArgumentOutOfRangeException();
            }
        }
    }
}

File: VacationCost/Java/src/com/itminds/VacationCostCalculator.java
package com.itminds;

import java.util.HashMap;

/**
 * Calculates the cost of vacation by using distance and transport method.
 */
public class VacationCostCalculator {
    private double distanceToDestination;
    private final TransportationMethod transportationMethod = new TransportationMethod();
    private final HashMap<String, Double> listofTransportation = transportationMethod.map();

    public VacationCostCalculator(final double disstanceToDestination) {
        this.distanceToDestination = disstanceToDestination;
    }

    /**
     * @param transportMethod string
     * @return result of the vacation cost
     */
    public double costOfVacation(final String transportMethod) {
        if (listofTransportation.containsKey(transportMethod)) {
            final double transportationCost = listofTransportation.get(transportMethod);
            return distanceToDestination * transportationCost;
        } else {
            throw new UnsupportedOperationException("Error: Transport Method " + transportMethod + " is not supported");
        }
    }

    /**
     * @return the distance of the destination
     */
    public double getDistanceToDestination() {
        return distanceToDestination;
    }

    /**
     * Set new distance of the destination
     *
     * @param distanceToDestination double
     */
    public void setDistanceToDestination(final double distanceToDestination) {
        this.distanceToDestination = distanceToDestination;
    }

}

